RIOT-OS/RIOT
# xtimer: check in xtimer_mutex_lock_timeout timeout callback if a thread blocked on mutex

This is the same pr as #10872 with a test and edited commit message.
it also includes a throwaway test commit.

### Contribution description

Prevent a possible race condition for xtimer_mutex_lock_timeout when _mutex_timeout fires just after the mutex was locked but before the xtimer was removed

The flow
```
int xtimer_mutex_lock_timeout(mutex_t *mutex, uint64_t timeout) {
   ...
    mutex_lock(mutex);
    <- mutex locked ->
    <- _mutex_timeout fires and tries to remove thread from mutex queue ->
    xtimer_remove(&t);
    ...
}
```

### Testing procedure

with the fix the test works, 
```
BOARD=native make -C tests/xtimer_mutex_lock_timeout/ flash test
...
starting test: xtimer mutex lock timeout
OK
> 
```

without the fix 
```
git revert HEAD
BOARD=native make -C tests/xtimer_mutex_lock_timeout/ flash test
...
starting test: xtimer mutex lock timeout
sys/xtimer/xtimer.c:239 => 0x56596856
*** RIOT kernel panic:
FAILED ASSERTION.

*** halted.


native: exiting
Unexpected end of file in expect script at "child.expect("OK")" (tests/xtimer_mutex_lock_timeout/tests/01-run.py:24)
```

### issues/PRs references

This is the same pr as #10872 with a test.
#11660: tracking pr for xtimer_mutex_lock_timeout function
